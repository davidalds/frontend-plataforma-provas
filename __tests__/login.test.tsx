import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import LoginPage from 'pages/auth/login'
import userEvent from '@testing-library/user-event'
import mockRouter from 'next-router-mock'
import server from '../mocks/server'

jest.mock('next/router', () => require('next-router-mock'))

// Testar logar sem informar dados aos campos

describe('<LoginPage />', () => {
  afterAll(() => {
    expect(render(<LoginPage />)).toMatchSnapshot()
  })

  beforeEach(() => {
    render(<LoginPage />)
  })

  test('login without pass data to the form', async () => {
    const btnLogin = screen.getByRole('button', { name: /Entrar/i })
    expect(btnLogin).toBeInTheDocument()

    // ACT
    await userEvent.click(btnLogin)

    const emailInput = screen.getByLabelText('E-mail:')
    const passwordInput = screen.getByLabelText('Senha:')

    expect(emailInput).toHaveAttribute('aria-invalid')
    expect(passwordInput).toHaveAttribute('aria-invalid')

    const messageErrors = screen.getAllByText('Campo obrigatÃ³rio')

    expect(messageErrors).toHaveLength(2)
  })

  test('test login inputs', async () => {
    const btnLogin = screen.getByRole('button', { name: /Entrar/i })
    expect(btnLogin).toBeInTheDocument()

    const emailInput = screen.getByLabelText('E-mail:')
    const passwordInput = screen.getByLabelText('Senha:')

    fireEvent.change(emailInput, { target: { value: 'user@gmail.com' } })
    fireEvent.change(passwordInput, { target: { value: 'pass1234' } })

    expect(emailInput).toHaveValue('user@gmail.com')
    expect(passwordInput).toHaveValue('pass1234')
  })
})
