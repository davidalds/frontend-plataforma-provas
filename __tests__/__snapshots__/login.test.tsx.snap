// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> test login inputs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1t53nb0"
      >
        <form
          class="chakra-stack css-5c4pa"
        >
          <h2
            class="chakra-heading css-13t2rws"
          >
            Plataforma Provas
          </h2>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-13ih5c7"
              for="field-:r4:"
              id="field-:r4:-label"
            >
              E-mail:
            </label>
            <input
              class="chakra-input css-0"
              id="field-:r4:"
              name="email"
              placeholder="Insira o e-mail"
              type="email"
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-13ih5c7"
              for="field-:r5:"
              id="field-:r5:-label"
            >
              Senha:
            </label>
            <input
              class="chakra-input css-0"
              id="field-:r5:"
              name="password"
              placeholder="Insira a senha"
              type="password"
            />
          </div>
          <button
            class="chakra-button css-uxt1e8"
            type="submit"
          >
            Entrar
          </button>
          <button
            class="chakra-button css-uxt1e8"
            type="button"
          >
            Novo Usuário
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1t53nb0"
    >
      <form
        class="chakra-stack css-5c4pa"
      >
        <h2
          class="chakra-heading css-13t2rws"
        >
          Plataforma Provas
        </h2>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-13ih5c7"
            for="field-:r4:"
            id="field-:r4:-label"
          >
            E-mail:
          </label>
          <input
            class="chakra-input css-0"
            id="field-:r4:"
            name="email"
            placeholder="Insira o e-mail"
            type="email"
          />
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-13ih5c7"
            for="field-:r5:"
            id="field-:r5:-label"
          >
            Senha:
          </label>
          <input
            class="chakra-input css-0"
            id="field-:r5:"
            name="password"
            placeholder="Insira a senha"
            type="password"
          />
        </div>
        <button
          class="chakra-button css-uxt1e8"
          type="submit"
        >
          Entrar
        </button>
        <button
          class="chakra-button css-uxt1e8"
          type="button"
        >
          Novo Usuário
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
